RF24Mesh.cpp

```diff
@@ -314,49 +314,47 @@ bool ESBMesh<network_t, radio_t>::requestAddress(uint8_t level)
 {
     RF24NetworkHeader header(MESH_MULTICAST_ADDRESS, NETWORK_POLL);
     //Find another radio, starting with level 0 multicast
-    IF_MESH_DEBUG(printf_P(PSTR("%u: MSH Poll\n"), millis()));
+    IF_MESH_DEBUG(printf_P(PSTR("MSH Poll Level %d\n"), level));
     network.multicast(header, 0, 0, level);
 
-    uint32_t timr = millis();
+    uint32_t pollTimeout = millis() + 55;
 #define MESH_MAXPOLLS 4
     uint16_t contactNode[MESH_MAXPOLLS];
     uint8_t pollCount = 0;
 
-    while (1) {
+    while (millis() < pollTimeout && pollCount < MESH_MAXPOLLS) {
 #if defined(MESH_DEBUG)
         bool goodSignal = radio.testRPD();
 #endif
-
         if (network.update() == NETWORK_POLL) {
-
-            memcpy(&contactNode[pollCount], &network.frame_buffer[0], sizeof(uint16_t));
-            if (pollCount > 0 && contactNode[pollCount] != contactNode[pollCount - 1]) { //Drop duplicate polls to help prevent duplicate requests
-                ++pollCount;
+            uint16_t contact = 0;
+            memcpy(&contact, &network.frame_buffer[0], sizeof(contact));
+
+            // Drop duplicate polls to help prevent duplicate requests
+            bool isDupe = false;
+            for (uint8_t i = 0; i < pollCount; ++i) {
+                if (contact == contactNode[i]) {
+                    isDupe = true;
+                    break;
+                }
             }
-            else {
+            if (!isDupe) {
+                contactNode[pollCount] = contact;
                 ++pollCount;
+                IF_MESH_DEBUG(printf_P(PSTR("MSH Poll %c -64dbm from 0%o \n"), (goodSignal ? '>' : '<'), contact));
```

Comment:
I'm not against having the statement here. Moving it out of this conditional block would hint at how many duplicates were encountered:
```
<number of signal statement> - <stated poll count> = <number of duplicates encountered>
```
Again, this isn't a requested change, just my thoughts.